base/channels.jl

```diff
@@ -242,10 +242,8 @@ Stacktrace:
 ```
 """
 function bind(c::Channel, task::Task)
-    # TODO: implement "schedulewait" and deprecate taskdone_hook
-    #T = Task(() -> close_chnl_on_taskdone(task, c))
-    #schedulewait(task, T)
-    register_taskdone_hook(task, tsk -> close_chnl_on_taskdone(tsk, c))
+    T = Task(() -> close_chnl_on_taskdone(task, c))
```

Comment:
The `close_chnl_on_taskdone` code had a weird internal structure to work around not being run in a proper task. I think that can be cleaned up now too.